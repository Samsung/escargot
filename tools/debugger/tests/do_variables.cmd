s
s
scope
variables
variables 1
s
s
s
s
scope
variables
variables 1
s
s
scope
variables
s
s
s
scope
variables
s
