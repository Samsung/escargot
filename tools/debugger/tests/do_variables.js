/*
 * Copyright (c) 2020-present Samsung Electronics Co., Ltd
 *
 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
 *  USA
 */

{
  let a = 6
  {
    const c = 8
    const d = 9
  }
  let b = 7
}

function f(a) {
  var b = 2
  let c
  return
}

f(1)

with({ p1:1, p2:2 }) {
    p1 = 3;
}

{
  // Too long identifier and string
  let a12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890 = 1,
      b = "12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890";
  b = 1
}
