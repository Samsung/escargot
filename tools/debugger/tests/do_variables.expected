Connecting to: localhost:6501
Connection created!!!
Stopped at tools/debugger/tests/do_variables.js:21
(escargot-debugger) s
Stopped at tools/debugger/tests/do_variables.js:23
(escargot-debugger) s
Stopped at tools/debugger/tests/do_variables.js:24
(escargot-debugger) scope
Declarative Environment
Declarative Environment
Global Environment
(escargot-debugger) variables
c: number: 8
d: unaccessible
(escargot-debugger) variables 1
a: number: 6
b: unaccessible
(escargot-debugger) s
Stopped at tools/debugger/tests/do_variables.js:26
(escargot-debugger) s
Stopped at tools/debugger/tests/do_variables.js:35
(escargot-debugger) s
Stopped at tools/debugger/tests/do_variables.js:30 (in f() at line:29, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_variables.js:32 (in f() at line:29, col:1)
(escargot-debugger) scope
Declarative Environment
Function Environment
Global Environment
(escargot-debugger) variables
c: undefined
(escargot-debugger) variables 1
a: number: 1
b: number: 2
(escargot-debugger) s
Stopped at tools/debugger/tests/do_variables.js:37
(escargot-debugger) s
Stopped at tools/debugger/tests/do_variables.js:38
(escargot-debugger) scope
Object Environment
Global Environment
(escargot-debugger) variables
p1: number: 1
p2: number: 2
(escargot-debugger) s
Stopped at tools/debugger/tests/do_variables.js:43
(escargot-debugger) s
Stopped at tools/debugger/tests/do_variables.js:44
(escargot-debugger) s
Stopped at tools/debugger/tests/do_variables.js:45
(escargot-debugger) scope
Declarative Environment
Global Environment
(escargot-debugger) variables
a1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567...: number: 1
b: string: 12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678...
(escargot-debugger) s
Connection closed.
